CommonTools/RecoAlgos/plugins/JetConstituentSelector.cc

```diff
@@ -28,76 +28,55 @@
 #include "CommonTools/UtilAlgos/interface/StringCutObjectSelector.h"
 #include "FWCore/Framework/interface/Event.h"
 
-template < class T, typename C = std::vector<typename T::ConstituentTypeFwdPtr> >
-class JetConstituentSelector : public edm::EDFilter {
-
+template <class T, typename C = std::vector<typename T::ConstituentTypeFwdPtr>>
+class JetConstituentSelector : public edm::stream::EDFilter<> {
```

Comment:
since "filter_ " is removed and the ::filter method return value is true, there is no more excuse to call this EDFilter.
Please change to EDProducer